src/lib/hyrise.hpp

```diff
@@ -2,26 +2,42 @@
 
 #include "boost/container/pmr/memory_resource.hpp"
 #include "concurrency/transaction_manager.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/topology.hpp"
 #include "storage/storage_manager.hpp"
 #include "utils/meta_table_manager.hpp"
 #include "utils/plugin_manager.hpp"
 #include "utils/singleton.hpp"
 
 namespace opossum {
 
+class JitRepository;
+class BenchmarkRunner;
+
 class Hyrise : public Singleton<Hyrise> {
  public:
   // Resets the Hyrise state by deleting its members (e.g., StorageManager) and
   // creating new ones. This is used especially in tests and can lead to a lot of
   // issues if there are still running tasks / threads that want to access a resource.
-  // You should be very sure that this is what you want. Have a look at base_test.hpp
-  // to see the correct order of resetting things.
-  static void reset() { get() = Hyrise{}; }
+  // You should be very sure that this is what you want.
+  static void reset() {
+    Hyrise::get().current_scheduler.get()->finish();
+    get() = Hyrise{};
+  }
+
+  void set_jit_repository(std::shared_ptr<JitRepository> repo) { jit_repository = repo; }
+
+  void set_benchmark_runner(std::shared_ptr<BenchmarkRunner> runner) { benchmark_runner = runner; }
 
   PluginManager plugin_manager;
   StorageManager storage_manager;
   TransactionManager transaction_manager;
   MetaTableManager meta_table_manager;
+  Topology topology;
+  CurrentScheduler current_scheduler;
+
+  std::shared_ptr<JitRepository> jit_repository;
+  std::shared_ptr<BenchmarkRunner> benchmark_runner;
```

Comment:
The Hyrise object should not be a co-owner of the benchmark runner. If whoever created it decides that it is done with it, it should be deleted. weak_ptr?